src/roi/Roi.ts

```diff
@@ -91,4 +145,513 @@ export class Roi {
+    }
+  }
+
+  get borderIDs() {
+    if (this.computed.borderIDs) {
+      return this.computed.borderIDs;
+    }
+    this._computeBorderIDs();
+    return this.computed.borderIDs || [];
+  }
+
+  get borderLengths(): number[] {
+    if (this.computed.borderLengths) {
+      return this.computed.borderLengths;
+    }
+    this._computeBorderIDs();
+    return this.computed.borderLengths || [];
+  }
+
+  get box() {
+    // points of the Roi that touch the rectangular shape
+    if (!this.computed.box) {
+      this.computed.box = getBox(this);
+    }
+    return this.computed.box;
+  }
+  /**
+   * Getter that calculates fill ratio of the ROI
+   */
+  get fillRatio() {
+    return this.surface / (this.surface + this.holesInfo.surface);
+  }
+  /**
+   * Getter that calculates sphericity of the ROI
+   */
+  get sphericity() {
+    return (2 * Math.sqrt(this.surface * Math.PI)) / this.perimeter;
+  }
+
+  /**
+   * Getter that calculates solidity of the ROI
+   */
+  get solidity() {
+    return this.surface / getConvexHull(this.getMask()).surface;
+  }
+
+  get roundness() {
+    /*Slide 24 https://static.horiba.com/fileadmin/Horiba/Products/Scientific/Particle_Characterization/Webinars/Slides/TE011.pdf */
+    return (
+      (4 * this.surface) /
+      (Math.PI * this.feretDiameters.maxDiameter.length ** 2)
+    );
+  }
+
+  get feretDiameters(): {
+    minDiameter: FeretDiameter;
+    maxDiameter: FeretDiameter;
+    aspectRatio: number;
+  } {
+    if (!this.computed.feretDiameters) {
+      this.computed.feretDiameters = getFeret(this.getMask());
+    }
+    return this.computed.feretDiameters;
+  }
+  /**
+   *
+   * @returns calculated properties as one object
+   */
+  toJSON() {
+    return {
+      id: this.id,
+      height: this.height,
+      width: this.width,
+      surface: this.surface,
+      eqpc: this.eqpc,
+      ped: this.ped,
+      feretDiameterMin: this.feretDiameters.minDiameter,
+      feretDiameterMax: this.feretDiameters.maxDiameter,
+      aspectRatio: this.feretDiameters.aspectRatio,
+      fillRatio: this.fillRatio,
+      sphericity: this.sphericity,
+      roundness: this.roundness,
+      solidity: this.solidity,
+      perimeter: this.perimeter,
+      perimeterInfo: this.perimeterInfo,
+      externalIDs: this.externalIDs,
+      externalLengths: this.externalLengths,
+      internalIDs: this.internalIDs,
+      borderIDs: this.borderIDs,
+      external: this.external,
+      borderLengths: this.borderLengths,
+      boxIDs: this.boxIDs,
+    };
+  }
+}
+/**
+ *
+ * @param roi -ROI
+ * @returns object which tells how many pixels are exposed externally to how many sides
+ */
+function getPerimeterInfo(roi: Roi) {
+  let roiMap = roi.getMap();
+  let data = roiMap.data;
+  let one = 0;
+  let two = 0;
+  let three = 0;
+  let four = 0;
+
+  for (let x = 0; x < roiMap.width; x++) {
+    for (let y = 0; y < roiMap.height; y++) {
+      let target = (y + roi.origin.row) * roiMap.width + x + roi.origin.column;
```

Comment:
I'm surprise that is is done on the full roiMap and not just target the specific ROI. This must be a bug and the current limiting step of image analysis.

Reply:
I'm surprise that is is done on the full roiMap and not just target the specific ROI. This must be a bug and the current limiting step of image analysis.